libs/utils/env.py

```diff
@@ -379,6 +379,14 @@ def _init_target(self, force = False):
             platform = Platform(model='MT8173')
             self.__modules = ['bl', 'cpufreq']
 
+        # Initialize gem5 simulation as a board
+        elif self.conf['board'].upper() == 'GEM5':
+            platform = devlib.platform.gem5.Gem5SimulationPlatform(name='gem5',
+                       gem5_bin=self.conf['gem5_bin'],
```

Comment:
What abut using a dictionary in target.conf

```python
my_conf = {
    ....
    "board" = "gem5",
    "gem5" = {
        "bin" = "...",
        "virtio" = "...",
        "args" = "...",
    }
```

And perhaps we can also "force" the things by assuming that if you have 'gem5' in conf then this is the board to use... but let see what @bjackman think about that ;-)